### PR TITLE
Update GitHub Actions versions

### PR DESCRIPTION
**Pull request by akx**
### Pull Request check-list

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

This PR bumps various GitHub Actions versions to current ones.

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -31,9 +31,8 @@ jobs:
      name: Code linters
      runs-on: ubuntu-latest
      steps:
-       - uses: actions/checkout@v2
-       - name: install python
```
Comment (dvora-h) [inline .github/workflows/integration.yaml:35]: @akx Why did you remove the name?

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -55,9 +54,8 @@ jobs:
        ACTIONS_ALLOW_UNSECURE_COMMANDS: true
      name: Python ${{ matrix.python-version }} ${{matrix.test-type}}-${{matrix.connection-type}} tests
      steps:
-       - uses: actions/checkout@v2
-       - name: install python
```
Comment (dvora-h) [inline .github/workflows/integration.yaml:59]: same

File: .github/workflows/integration.yaml [hunk 3]
```diff
@@ -94,9 +92,8 @@ jobs:
       matrix:
         extension: ['tar.gz', 'whl']
     steps:
-      - uses: actions/checkout@v2
-      - name: install python
```
Comment (dvora-h) [inline .github/workflows/integration.yaml:98]: same

File: .github/workflows/integration.yaml [hunk 4]
```diff
@@ -110,9 +107,8 @@ jobs:
       matrix:
         python-version: ['3.7', '3.8', '3.9', '3.10', 'pypy-3.7']
     steps:
-      - uses: actions/checkout@v2
-      - name: install python ${{ matrix.python-version }}
```
Comment (dvora-h) [inline .github/workflows/integration.yaml:114]: same

File: .github/workflows/integration.yaml (hunk 1 above)
Comment (akx) [reply .github/workflows/integration.yaml:35]: Not all of the `actions/setup-python` steps in the YAML files had a name beforehand, anyway.

I see no actual value in a longer YAML file with the only visible change being "install python" being shown instead of "Run actions/setup-python@v4" in logs.

File: .github/workflows/integration.yaml (hunk 2 above)
Comment (akx) [reply .github/workflows/integration.yaml:59]: Not all of the `actions/setup-python` steps in the YAML files had a name beforehand, anyway.

I see no actual value in a longer YAML file with the only visible change being "install python" being shown instead of "Run actions/setup-python@v4" in logs.

File: .github/workflows/integration.yaml (hunk 3 above)
Comment (akx) [reply .github/workflows/integration.yaml:98]: See above.

File: .github/workflows/integration.yaml (hunk 4 above)
Comment (akx) [reply .github/workflows/integration.yaml:114]: See above.

**Review by dvora-h — APPROVED** [review]
